.github/workflows/validate.yml

```diff
@@ -0,0 +1,78 @@
+# Based on https://github.com/actions-rs/meta/blob/master/recipes/quickstart.md
+
+on:
+  push: { }
+
+name: Validate
+
+jobs:
+  check:
+    name: Check
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ windows-latest, ubuntu-latest, macOS-latest ]
+        rust: [ stable ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: ${{ matrix.rust }}
+
+      - name: Run cargo check
+        run: cargo check
+
+  test:
+    name: Test Suite
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Run cargo test
+        run: cargo test
+
+  test-windows:
+    name: Test Suite Windows
+    runs-on: windows-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Run cargo test
+        run: cargo test
```

Comment:
I would say windows support is extremely important if this is supposed to eventually support a full-node wallet.

Reply:
That's kind of the question. If we aim for Windows support _eventually_, I believe it's a good idea to start early as a late support for Windows is a complete mess. 

On the other hand, supporting Windows is a pain in the ***, and I don't want us to lose days struggling with fixing Windows builds while we have better stuff to do. So that's a broader discussion we need to have with the rest of the maintainers and record eventually as an EDR. For now, if it doesn't hurt to have it then let's have it.